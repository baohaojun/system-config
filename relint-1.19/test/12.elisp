;;; Relint test file 12          -*- emacs-lisp -*-

;; Test auto-mode-alist

(define-generic-mode my-mode
  nil
  nil
  nil
  '(".aa\\'" "\\.bb$" "^cc.*dd"))

(add-to-list 'auto-mode-alist '("\\.ee$" . some-mode))
(push '(".ff\\'" . some-mode) auto-mode-alist)
(setq auto-mode-alist (cons '(".gg\\'" . some-mode) auto-mode-alist))
(setq auto-mode-alist (append '((".hh\\'" . some-mode)
                                (".ii\\'" . some-mode))
                              auto-mode-alist))

;; File-matching regexp functions

(defun f12 (d)
  (directory-files d nil ".txt\\'")
  (directory-files-and-attributes d nil "\\.pas$")
  (directory-files-recursively d "^abc")
  (modify-coding-system-alist 'file "\\.ml$" 'utf-8)
  (modify-coding-system-alist 'process "+xx$" 'utf-8))
