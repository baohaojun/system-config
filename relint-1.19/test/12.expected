12.elisp:9:6: In define-generic-mode my-mode: Possibly unescaped `.' in file-matching regexp (pos 0)
  ".aa\\'"
   ^
12.elisp:9:20: In define-generic-mode my-mode: Use \' instead of $ in file-matching regexp (pos 4)
  "\\.bb$"
   .....^
12.elisp:9:24: In define-generic-mode my-mode: Use \` instead of ^ in file-matching regexp (pos 0)
  "^cc.*dd"
   ^
12.elisp:11:39: In add-to-list: Use \' instead of $ in file-matching regexp (pos 4)
  "\\.ee$"
   .....^
12.elisp:12:10: In auto-mode-alist: Possibly unescaped `.' in file-matching regexp (pos 0)
  ".ff\\'"
   ^
12.elisp:13:32: In auto-mode-alist: Possibly unescaped `.' in file-matching regexp (pos 0)
  ".gg\\'"
   ^
12.elisp:14:35: In auto-mode-alist: Possibly unescaped `.' in file-matching regexp (pos 0)
  ".hh\\'"
   ^
12.elisp:15:35: In auto-mode-alist: Possibly unescaped `.' in file-matching regexp (pos 0)
  ".ii\\'"
   ^
12.elisp:21:27: In call to directory-files: Possibly unescaped `.' in file-matching regexp (pos 0)
  ".txt\\'"
   ^
12.elisp:22:48: In call to directory-files-and-attributes: Use \' instead of $ in file-matching regexp (pos 5)
  "\\.pas$"
   ......^
12.elisp:23:35: In call to directory-files-recursively: Use \` instead of ^ in file-matching regexp (pos 0)
  "^abc"
   ^
12.elisp:24:43: In call to modify-coding-system-alist: Use \' instead of $ in file-matching regexp (pos 4)
  "\\.ml$"
   .....^
12.elisp:25:41: In call to modify-coding-system-alist: Unescaped literal `+' (pos 0)
  "+xx$"
   ^
